src/cirq_iqm/devices/iqm_device.py

```diff
@@ -238,7 +238,7 @@ def route_circuit(
         new_measurements = []
         for _, op in measurement_ops:
             new_qubits = [final_qubit_mapping[q] for q in op.qubits]
-            new_measurement = cirq.measure(*new_qubits, key=op.gate.key)
+            new_measurement = cirq.measure(*new_qubits, key=op.gate.key)  # type: ignore
```

Comment:
Not sure there is a good way around this. The operations that end up in the `measurement_ops` are measurement operations, so they must have a measurement key. However, mypy cannot deduce this and complains about `key` not being found in `op.gate`.

Reply:
`findall_operations_with_gate_type` returns a tuple of three items where the last is operation gate casted to the requested gate type (`MeasurementGate`). That one should be recognized to have `.key`, but yeah, this will require modifying `measurement_ops` all over the place.

Reply:
Thanks, that turned out to be a minor update. Check out the latest change please.